flox/core.py

```diff
@@ -550,6 +562,160 @@ def chunk_argreduce(
 
 
 def chunk_reduce(
+    array: np.ndarray,
+    by: np.ndarray,
+    func: T_Funcs,
+    expected_groups: pd.Index | None,
+    axis: T_AxissOpt = None,
+    fill_value: T_FillValues = None,
+    dtype: T_Dtypes = None,
+    reindex: bool = False,
+    engine: T_Engine = "numpy",
+    kwargs: Sequence[dict] | None = None,
+    sort: bool = True,
+) -> IntermediateDict:
+    """
+    Wrapper for numpy_groupies aggregate that supports nD ``array`` and
+    mD ``by``.
+
+    Core groupby reduction using numpy_groupies. Uses ``pandas.factorize`` to factorize
+    ``by``. Offsets the groups if not reducing along all dimensions of ``by``.
+    Always ravels ``by`` to 1D, flattens appropriate dimensions of array.
+
+    When dask arrays are passed to groupby_reduce, this function is called on every
+    block.
+
+    Parameters
+    ----------
+    array : numpy.ndarray
+        Array of values to reduced
+    by : numpy.ndarray
+        Array to group by.
+    func : str or Callable or Sequence[str] or Sequence[Callable]
+        Name of reduction or function, passed to numpy_groupies.
+        Supports multiple reductions.
+    axis : (optional) int or Sequence[int]
+        If None, reduce along all dimensions of array.
+        Else reduce along specified axes.
+
+    Returns
+    -------
+    dict
+    """
+
+    if not (isinstance(func, str) or callable(func)):
```

Comment:
I llike this but we shoudl do this at the `groupby_reduce` level. Otherwise for dask stuff it will only raise at compute time. Unless you're checking that we've created `Aggregation` objects correctly.

Reply:
This is to make sure `funcs` is always a sequence. `chunk_reduce` is used on too many places though, it would require too many unrelated typing changes. I think it's for another PR once the mypy CI is active.